### ISSUE TITLE
Deployment fails with no error message

### ISSUE THREAD

**Issue by dragosiordachioaia** [aws-amplify/amplify-console#1039]
I added Cypress tests to my build process about a week ago, and initially it worked. Then I added visual snapshots to my tests, which produce .png artifacts, as well as the .mp4 ones normally generated by Cypress. I am skipping these tests on my master branch (via a Node.js script), but the snapshots are always there. This leads to 2 situations:
- PR previews run the tests ok, but fail on deployment - no error message, it just stops after uploading the test artifacts
- master branch skips tests, but also fails on deployment - again no error message, it just stops after uploading the test artifacts

Example from the build output for master:

`2020-09-10T20:21:11 [INFO]: Starting Deployment
2020-09-10T20:21:11 [INFO]: Updating Edge config
2020-09-10T20:21:12 [INFO]: Start deploying test artifact
2020-09-10T20:21:12 [INFO]: Deploying snapshots/login.spec.js/login-empty.snap.png
2020-09-10T20:21:12 [INFO]: Deploying snapshots/review.spec.js/annotation-element.snap.png
2020-09-10T20:21:12 [INFO]: Deploying snapshots/review.spec.js/square-freehand.snap.png
2020-09-10T20:21:12 [INFO]: Finish deploying test artifact`

If I remove the "test" step from the build entirely, it works just fine. This is really concerning, since a significant part of CI is to run tests, and it's exactly what I cannot do.

**Comment by ambriglia** [comment]
Also, as a workaround for this, you can comment out the files block in the amplify.yml file and your tests will still run, but no files will attempt to be uploaded. See image below. What is in red is what we removed.

![image](https://user-images.githubusercontent.com/9729776/93232766-4674ff00-f748-11ea-9e6e-2d36e833fa23.png)

**Comment by Athena96** [comment]
Hi @dragosiordachioaia can you please share your appId

**Comment by dragosiordachioaia** [comment]
the appId is d18apgfog4c8xo

**Comment by dragosiordachioaia** [comment]
@ambriglia yes, I this started happening all of a sudden

**Comment by Athena96** [comment]
Hi @dragosiordachioaia are you still experiencing this issue? I'm not able to repro on my end. 
Can you please share the file structure of your app (including cypress tests). You can share here or email aws-amplify-customer@amazon.com
Thank you!

**Comment by kahdojay** [comment]
@ambriglia can you share your region, appid, branch name, and example build number that exhibits deploy failure behavior?

**Comment by kahdojay** [comment]
@dragosiordachioaia which build number is the screenshot you posted above?

**Comment by kahdojay** [comment]
@dragosiordachioaia can you retry in eu-west-2 and let us know if you're still seeing the issues with both master branch deploys and pr preview deploys?

**Comment by ambriglia** [comment]
@kahdojay I already have an internal AWS support ticket open with all the appropriate details.

d2df4ylja65m9v
us-east-2
master
324

I currently have the workaround still implemented I referenced above so I am not sure if we are still experiencing the issue.

**Comment by kahdojay** [comment]
@ambriglia thanks for the details and pointing me to the support case as well. Will provide more details there but in short I think you're having the same issue as #115, we pushed a fix for us-east-2 and it looks like you've had a successful deploy since, but let me know if you're still seeing the same issue.

**Comment by kahdojay** [comment]
@ambriglia thanks for the details and pointing me to the support case as well. Will provide more details there but in short I think you're having the same issue as #115, we pushed a fix for us-east-2 yesterday and it looks like you've had a successful deploy since, but let me know if you're still seeing the same issue.

**Comment by gspeicher** [comment]
i am having similar issues here.

appid: d2r3gwm433d7n4
region: us-east-1
branch: master
build: 111
deploy output:
```
2020-09-25T12:11:54 [INFO]: Starting Deployment
2020-09-25T12:11:55 [INFO]: Updating Edge config
2020-09-25T12:11:55 [INFO]: Start deploying test artifact
2020-09-25T12:11:55 [INFO]: Deploying output.json
2020-09-25T12:11:55 [INFO]: Finish deploying test artifact
[long pause here; one hour of time elapsed and the console was cleared and output restarted at the top]
2020-09-25T13:11:55 [INFO]: Beginning deployment for application d2r3gwm433d7n4, branch:master, buildId 0000000111
2020-09-25T13:11:55 [WARN]: Artifact bucket not found for application d2r3gwm433d7n4. Exiting
2020-09-25T13:11:55 [ERROR]: Deployment terminated unsuccessfully
```
then it just hangs indefinitely. it is still stuck there as i write this. everything up to the deploy ran successfully. i had been running with the entire test block removed from amplify.yml (as a workaround to this bug, which i ran into a few weeks ago), and until i re-enabled the tests, the web app had been deploying successfully in just a few seconds.

the last successful deploy was build 106:
```
2020-09-23T16:17:33 [INFO]: Beginning deployment for application d2r3gwm433d7n4, branch:master, buildId 0000000106
2020-09-23T16:17:35 [INFO]: Got archive: 77231708 bytes
2020-09-23T16:17:35 [INFO]: Archive unzipped: 49 files
2020-09-23T16:17:37 [INFO]: Deployment complete
```

as i mentioned above, i ran into this problem a few weeks ago, but I would also note that the behavior has changed as an apparent result of your recent changes. when it was failing back at the beginning of september, the deployment output was:
```
2020-09-02T19:55:05 [INFO]: Starting Deployment
2020-09-02T19:55:05 [ERROR]: Failed to deploy
2020-09-02T19:55:05 [INFO]: Deployment finished.
```

so it seems like it is moving in the right direction, but not quite there yet. i will add that i am a little bit confused about what is the expected behavior when the build artifact folder is different than the test artifact folder. do they both get copied to the root of the S3 bucket? i committed a subsequent change to my amplify.yml file to put the test results in the same directory as the build artifacts, and removed the test artifact section from the config, to see if that makes a difference. however, that build (112) seems to be stuck between the provision and build steps, presumably because build 111 is still stuck and holding the resources necessary for the build.

**Comment by kahdojay** [comment]
@gspeicher can you share your amplify.yml that you included in your repo for build 111 as well as the version for 112? You can share here or email us at aws-amplify-customer@amazon.com

**Comment by gspeicher** [comment]
I just emailed them. I should have mentioned in the email that build 112 (and 113) eventually ran and failed with the error:

`2020-09-25T21:17:44.097Z [ERROR]: !!! Base Directory not specified for artifacts, unable to create test artifact.`

I just reverted amplify.yml to the same version that was used in build 111, so build 114 is under way now. Let me know if there is anything else I can do or provide. Thanks!

**Comment by gspeicher** [comment]
Update: as of last night, the build is failing differently again. Build 118 is using the same amplify.xml as build 111, but the deploy is failing differently than the build 111 deploy did:

```
2020-09-28T19:23:21 [INFO]: Starting Deployment
2020-09-28T19:23:21 [ERROR]: Failed to deploy
```

This is more like the failure on 9/2 except now it does not output "Deployment Finished" as it did back then.

**Comment by gspeicher** [comment]
I see a lot of activity on #115 but this problem is still unchanged for me. Our Amplify project is stuck in the Deploy phase of the pipeline as of yesterday:

`2020-10-27T15:10:31 [INFO]: Starting Deployment
2020-10-27T15:10:31 [INFO]: Updating Edge config
2020-10-27T15:10:32 [INFO]: Start deploying test artifact
2020-10-27T15:10:32 [INFO]: Deploying output.json
2020-10-27T15:10:32 [INFO]: Finish deploying test artifact`

Any updates on this issue?

**Comment by litwicki** [comment]
@gspeicher are you still experiencing this issue?

**Comment by gspeicher** [comment]
Yes — in fact I got tired of waiting and finally disabled the test block in
amplify.xml again yesterday (Dec 7th) to get our first successful
build+deploy since late October. After committing the update to
amplify.xml, I had to kill the previous build that was stuck in the Deploy
phase since Nov 23rd.

**Comment by kshitijcode** [comment]
This issue is still happening for me as well.

**Comment by Charlie059** [comment]
`2023-08-11T18:45:54 [INFO]: Starting Deployment
2023-08-11T18:45:55 [INFO]: Start deploying test artifact
2023-08-11T18:45:55 [INFO]: Deploying videos/interveiw/tutorialFlow.cy.js.mp4
2023-08-11T18:45:55 [INFO]: Deploying videos/interveiw/tutorialForm.cy.js.mp4
2023-08-11T18:45:55 [INFO]: Deploying videos/registor2Login/registor2Login.cy.js.mp4
2023-08-11T18:45:55 [INFO]: Finish deploying test artifact
2023-08-11T18:45:55 [INFO]: Updating Edge config`
